maintainers/scripts/check-unstable-version.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p git nix
+
+set -o errexit -o nounset -o pipefail
+shopt -s extglob
+
+if [[ "$#" -eq 0 ]]; then
+    echo "Usage: $0 [PR number] attrPath..." >&2
+    echo "Attribute paths are optional only if PR is specified" >&2
+    exit 1
+fi
+
+nixpkgsRemote=https://github.com/NixOS/nixpkgs.git
+
+pr=
+if [[ "$1" =~ ^[1-9] ]]; then
+    pr=$1
+    shift
+fi
+attrPaths=("$@")
+
+cleanupHooks=()
+function cleanup {
+    for ((i = ${#cleanupHooks[@]} - 1; i >= 0; i--)); do
+        ${cleanupHooks[$i]}
+    done
+}
+trap cleanup EXIT
+
+function mkWorktree {
+    worktree=$(mktemp -d)
+    cleanupHooks+=(rmWorktree)
+}
+
+function rmWorktree {
+    cd
+    rm -rf "$worktree"
+}
+
+function mkGitWorktree {
+    git worktree add --quiet -d "$worktree" "$1"
+    cleanupHooks+=(rmGitWorktree)
+}
+
+function rmGitWorktree {
+    git worktree remove -f "$worktree"
+}
+
+function mkBareSrcDir {
+    bareSrcDir=$(mktemp -u)
+    cleanupHooks+=(rmBareSrcDir)
+}
+
+function rmBareSrcDir {
+    rm -rf "$bareSrcDir"
+}
+
+if [[ "$pr" ]]; then
+    mkWorktree
+    git fetch --quiet "$nixpkgsRemote" "pull/$pr/head"
+    mkGitWorktree FETCH_HEAD
+    cd "$worktree"
+    mkdir .src
```

Comment:
Creating a new worktree takes quite a bit of time on my machines. It would probably be a good idea to have an option to just use the current worktree.